src/renderer/src/routes/(MapTabs)/_Map.tsx

```diff
@@ -1,34 +1,99 @@
-import { Paper } from '@mui/material'
+import { Suspense } from 'react'
+import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined'
+import PostAddIcon from '@mui/icons-material/PostAdd'
+import SettingsIcon from '@mui/icons-material/Settings'
+import { Box, CircularProgress, Paper } from '@mui/material'
 import Tab from '@mui/material/Tab'
 import Tabs from '@mui/material/Tabs'
+import { styled } from '@mui/material/styles'
 import { Outlet, createFileRoute, useNavigate } from '@tanstack/react-router'
+import { defineMessages, useIntl } from 'react-intl'
 
+import { VERY_LIGHT_GREY, WHITE } from '../../colors'
+import { Text } from '../../components/Text'
 import type { FileRoutesById } from '../../routeTree.gen'
 
+const m = defineMessages({
+	setting: {
+		id: 'tabBar.label.settings',
+		defaultMessage: 'Settings',
+	},
+	about: {
+		id: 'tabBar.label.about',
+		defaultMessage: 'About',
+	},
+})
+
+const Container = styled('div')({
+	display: 'flex',
+	backgroundColor: WHITE,
+	height: '100%',
+})
+
+const MapTabStyled = styled(MapTab)({
+	width: 60,
+})
+
 export const Route = createFileRoute('/(MapTabs)/_Map')({
 	component: MapLayout,
 })
 
 export function MapLayout() {
 	const navigate = useNavigate()
-	const renderCount = React.useRef(0)
-	renderCount.current = renderCount.current + 1
+	const { formatMessage } = useIntl()
 	return (
-		<div>
-			<Tabs
-				onChange={(_, value) => navigate({ to: value as MapTabRoute })}
-				orientation="vertical"
-			>
-				<MapTab label="Tab 1" value={'/tab1'} />
-				<MapTab label="Tab 2" value={'/tab2'} />
-			</Tabs>
-			<Paper>
-				<Outlet />
+		<Container>
+			<Paper elevation={3} sx={{ display: 'flex' }}>
+				<Tabs
+					sx={{ '& .MuiTabs-flexContainer': { height: '100%' } }}
+					onChange={(_, value) => navigate({ to: value as MapTabRoute })}
+					orientation="vertical"
+				>
+					{/* PostAddIcon is a placeholder icon */}
+					<MapTabStyled
+						data-testid="tab-observation"
+						icon={<PostAddIcon />}
+						value={'/tab1'}
+					/>
+					<Box flexGrow={1} />
+
+					<MapTabStyled
+						icon={<SettingsIcon />}
+						label={
+							<Text style={{ fontSize: 10 }} kind="title">
+								{formatMessage(m.setting)}
+							</Text>
+						}
+						value={'/tab2'}
+					/>
+					<MapTabStyled
+						icon={<InfoOutlinedIcon />}
+						label={
+							<Text style={{ fontSize: 10 }} kind="title">
+								{formatMessage(m.about)}
+							</Text>
+						}
+						value={'/tab2'}
+					/>
+				</Tabs>
+
+				<Box
+					sx={{
+						width: 300,
+						borderLeftColor: VERY_LIGHT_GREY,
+						borderLeftWidth: '1px',
+						borderLeftStyle: 'solid',
+					}}
+				>
+					<Suspense fallback={<CircularProgress />}>
+						<Outlet />
+					</Suspense>
+				</Box>
 			</Paper>
-			<div>map component here</div>
-			<div>parent map component render count: {renderCount.current}</div>
-		</div>
+			<Suspense fallback={<CircularProgress />}>
+				<div>map component here</div>
+			</Suspense>
```

Comment:
This component seems very complex. It handles routing, tab rendering, Suspense and layout. Could there be a separate function or hook to handle the logic of determining which tab is selected?
It would be nice if MapLayout could just focus on Layout

Reply:
Im open to simplfying this. Im just not fully sure how we do that. Do you have any suggestions?

Reply:
Maybe a useMapTabs hook to determine what tab to show? But maybe that is unnecessary. Just a suggestion.

Reply:
So, because the tabs needed a `value` prop it had to be aware of the navigation name. This requires the use of `useLocation` hook. This was causing the entire layout screen and all its children (aka the whole app) to rerender on every change in location. So I had to encapsulate the tabs into its own component. I think it simplifies the component?

I think this brings up the question of what is the scope of a component. Gregor has argued that the parent component should contain all the logic, and while it is messier, it does expose everything. And while i don't fully agree, he has some good points. I think there needs to be a more in depth conversation about what are the standards for components.

Regardless, I think the optimization of encapsulating the logic makes sense here because navigation touches everything, and this is a layout component that is the parent for the rest of the app. So the optimization is a necessary as opposed to an over optimization.

Reply:
I think it does too. Thanks. I think we will be glad of this later.